pytorch_lightning/utilities/parsing.py

```diff
@@ -204,46 +204,57 @@ def save_hyperparameters(
     obj: Any, *args: Any, ignore: Optional[Union[Sequence[str], str]] = None, frame: Optional[types.FrameType] = None
 ) -> None:
     """See :meth:`~pytorch_lightning.LightningModule.save_hyperparameters`"""
-
+    # empty container
     if len(args) == 1 and not isinstance(args, str) and not args[0]:
-        # args[0] is an empty container
         return
-
-    if not frame:
-        current_frame = inspect.currentframe()
-        # inspect.currentframe() return type is Optional[types.FrameType]: current_frame.f_back called only if available
-        if current_frame:
-            frame = current_frame.f_back
-    if not isinstance(frame, types.FrameType):
-        raise AttributeError("There is no `frame` available while being required.")
-
-    if is_dataclass(obj):
-        init_args = {f.name: getattr(obj, f.name) for f in fields(obj)}
-    else:
-        init_args = get_init_args(frame)
-    assert init_args, "failed to inspect the obj init"
-
-    if ignore is not None:
-        if isinstance(ignore, str):
-            ignore = [ignore]
-        if isinstance(ignore, (list, tuple)):
-            ignore = [arg for arg in ignore if isinstance(arg, str)]
-        init_args = {k: v for k, v in init_args.items() if k not in ignore}
-
-    if not args:
-        # take all arguments
-        hp = init_args
-        obj._hparams_name = "kwargs" if hp else None
+    # container
+    elif len(args) == 1 and isinstance(args[0], (Namespace, dict)):
+        # todo: omegaconf?
```

Comment:
Mind adressing this comment in this PR ?

Reply:
Overlooked this when I was doing the bug fixes sorry about that.

I'm hesitant to import omegaconf here (to add to `isinstance`), while the previous implementation only checks for `len(args) == 1` and `not isinstance(args[0], str)` which is quite ambiguous. Perhaps there is another check I missed? or is it fine to import omegaconf here?